hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/KafkaConsumerGroupManager.java

```diff
@@ -0,0 +1,111 @@
+                    .runAsync(new CreateConsumerGroupTask(subscription, partition), executor)
+                    .whenComplete((done, e) -> {
+                        if (e == null) {
+                            logger.info("Successfully created consumer group for subscription {} on partition {}, cluster: {}",
+                                    subscription.getQualifiedName(), partition, clusterName);
+                        } else {
+                            logger.error("Failed to create consumer group for subscription {} on partition {}, cluster: {}",
+                                    subscription.getQualifiedName(), partition, clusterName);
+                        }
+                    })
+        );
+    }
+
+    class CreateConsumerGroupTask implements Runnable {
+
+        private final Logger logger = LoggerFactory.getLogger(CreateConsumerGroupTask.class);
+
+        private final Subscription subscription;
+        private final int partition;
+        private final TopicPartition topicPartition;
+
+        CreateConsumerGroupTask(Subscription subscription, int partition) {
+            this.subscription = subscription;
+            this.partition = partition;
+            this.topicPartition = new TopicPartition(subscription.getQualifiedTopicName(), partition);
+        }
+
+        @Override
+        public void run() {
+            logger.info("Creating consumer group for subscription {} on partition {}", subscription.getQualifiedName(), partition);
+
+            int leaderId = brokerStorage.readLeaderForPartition(new TopicAndPartition(topicPartition));
+            ConsumerGroupId groupId = kafkaNamesMapper.toConsumerGroupId(subscription.getQualifiedName());
+            KafkaConsumer<byte[], byte[]> kafkaConsumer = createKafkaConsumer(groupId, leaderId);
+
+            kafkaConsumer.assign(ImmutableSet.of(topicPartition));
+            long offset = kafkaConsumer.position(topicPartition);
+            kafkaConsumer.commitSync(ImmutableMap.of(topicPartition, new OffsetAndMetadata(offset)));
+            kafkaConsumer.close();
+        }
+
+        private KafkaConsumer<byte[], byte[]> createKafkaConsumer(ConsumerGroupId groupId, int leaderId) {
+            BrokerDetails brokerDetails = brokerStorage.readBrokerDetails(leaderId);
+            String bootstrapServers = brokerDetails.getHost() + ":" + brokerDetails.getPort();
+            return new KafkaConsumer<>(properties(groupId, bootstrapServers));
+        }
+
+        private Properties properties(ConsumerGroupId groupId, String bootstrapServers) {
+            Properties props = new Properties();
+            props.put(BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+            props.put(GROUP_ID_CONFIG, groupId.asString());
+            props.put("key.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+            props.put("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+            return props;
```

Comment:
you should include `ENABLE_AUTO_COMMIT_CONFIG`=false

Reply:
Done